crates/wasi-provider/src/states/running.rs

```diff
@@ -73,17 +27,22 @@ impl State<PodState> for Running {
 
         while let Some((name, status)) = pod_state.run_context.status_recv.recv().await {
             // TODO: implement a container state machine such that it will self-update the Kubernetes API as it transitions through these stages.
-            if let Err(e) = patch_container_status(&client, &pod.name(), name, &status).await {
+
+            if let Err(e) = patch_container_status(&client, &pod, &name, &status, false).await {
                 error!("Unable to patch status, will retry on next update: {:?}", e);
             }
+
             if let Status::Terminated {
                 timestamp: _,
                 message,
                 failed,
             } = status
             {
                 if failed {
-                    return Transition::next(self, Error { message });
+                    // This appears to be required by the test `test_module_exiting_with_error`
```

Comment:
I don't agree with `fail_fatal` here or in the init pod state, but it appears to be required by the integration tests. The tests expect the Pod to fail whereas I would expect it to go to `Error`. Incidentally there appears to be a bug when the Pod does go to `Error` and then back to `Registered`, the wasm runtime encounters some error with a channel. We may need to do some clean up in the error state in anticipation of going back to `Registered`.

Reply:
Would it be all right if we pulled this out into a separate PR? That way we can have this discussion separate from the container status patching logic.

Reply:
There was discussion about this on the 'remove panics from handlers' PR - this was to replicate existing behaviour which was relied on by the integration tests.  I think everyone agrees that it is not the desired behaviour, but changing the behaviour should be tackled as a specific work item (which will be non-trivial), not rolled in along with other work.

Reply:
Never mind, I see you already raised an issue to do this as a separate item.  Thanks!